### PR TITLE
fix: upload fixes

### PR DESCRIPTION
**Pull request by dreinon**

### CODE REVIEW

**Review by anand2312 — APPROVED** [review]
Tests are failing but the changes seem fine, although I haven't tested them locally to see if everything still works. If you can verify that go ahead and merge this 🚀

**Comment by dreinon** [comment]
@anand2312 right, I don't really know why it fails when testing sync. It works locally, we might have to sleep a couple seconds after upload for the download method to find it somehow.

File: tests/_async/test_client.py [hunk 1]
```diff
@@ -119,14 +121,18 @@ async def test_client_upload_file(
...
     file_path = file["local_path"]
     mime_type = file["mime_type"]
+    file_content = file["file_content"]
     bucket_file_path = file["bucket_path"]
     bucket_folder = file["bucket_folder"]
     options = {"content-type": mime_type}
 
     await storage_file_client.upload(bucket_file_path, file_path, options)
+
+    sleep(3)
```
Comment (anand2312) [inline tests/_async/test_client.py:131]: Why is this not an `asyncio.sleep`?

Comment (dreinon) [reply tests/_async/test_client.py:131]: Hi! Since it has to be compatible with sync.

**Pull request merged**